src/hooks/useSearchParams.tsx

```diff
@@ -25,12 +25,18 @@ export const useSearchParams = (): [MappedSearchParams, SetURLSearchParams] => {
   const [searchParams, internalSetSearchParams] = React.useState(toMap(location.search));
 
   /**
-   * Update and add to the existing search parameters on the URL and the internal state.
+   * Change existing search parameters on the URL and the internal state.
+   * Operations:
+   * 1. ADD - by adding a new key:value pair to the object
+   * 2. UPDATE - provide a new value for an exisitng key
+   * 3. DELETE - set value 'undefined' for an existing key
    *
    * @param {MappedSearchParams} params
    */
   const updateSearchParams: SetURLSearchParams = (params) => {
-    const combinedPrams = { ...searchParams, ...params };
+    const combinedPrams = Object.fromEntries(
+      Object.entries({ ...searchParams, ...params }).filter(([name, value]) => value !== undefined),
+    );
```

Comment:
Just removing keys with a value of `undefined`?  Maybe a little less voodoo :wink: ...

```js
for (const key in combinedParams) if (combinedParams[key] === undefined) delete combinedParams[key]
```

Reply:
`delete` operator is used so rarely that a causal reader of this code may panic :)

The approach with filtering was my first thought. It's a bit verbose but mutation-free.

Reply:
Or just a helper function to do it...